src/java/org/apache/cassandra/batchlog/Batch.java

```diff
@@ -113,6 +120,22 @@ public void serialize(Batch batch, DataOutputPlus out, int version) throws IOExc
             }
         }
 
+        @VisibleForTesting
+        public void reserialize(Batch batch, DataOutputPlus out, int version) throws IOException
+        {
+            assert !batch.isLocal() : "attempted to reserialize a 'local' batch";
+
+            UUIDSerializer.serializer.serialize(batch.id, out, version);
+            out.writeLong(batch.creationTime);
+
+            out.writeUnsignedVInt(batch.encodedMutations.size());
+            
+            for (ByteBuffer mutation : batch.encodedMutations)
+            {
+                out.write(mutation);
+            }
+        }
```

Comment:
This block now exists only because of the in-JVM [test framework](https://github.com/apache/cassandra/pull/821/commits/a715388a1c7029f5a76e0061715856fe1c3bb391#diff-321b52220c5bd0aaadf275a845143eb208c889c2696ba0d48a5fc880551131d8R313). I could be persuaded to do something less invasive, like perhaps just exposing the encoded mutations, given the other elements are already accessible. (i.e. The `reserialize()` method would move to `Instance`.)

CC @ifesdjeen @adelapena

Reply:
Indeed exposing the encoded mutations sounds less aggressive, although I'm not against the test method. If we preserve the `reserialize` method we should add some JavaDoc indicating why it exists.

Reply:
This is where I'd be interested to know what our second reviewer thinks ;)

Reply:
See https://github.com/apache/cassandra/pull/821/files?file-filters%5B%5D=.java#diff-321b52220c5bd0aaadf275a845143eb208c889c2696ba0d48a5fc880551131d8R339

@adelapena @ekaterinadimitrova2 Given the only user is `Instance` at this point, I'm fine with just exposing the encoded mutations, if I can get that to work.